lib/socketry/exceptions.rb

```diff
@@ -0,0 +1,18 @@
+module Socketry
+  # Generic catch all for all Socketry errors
+  SocketError = Class.new(IOError)
```

Comment:
IMHO if one goes `MyClass = Class.new(ParentClass)`, he must also use that style everywhere else:

``` ruby
module Socketry
  Resolver = Module.new do
    Error = Class.new(AddressError)
  end
end
```